client/src/components/UserAvatar/index.js

```diff
@@ -30,9 +31,13 @@ class UserAvatar extends PureComponent {
       const { filename } = props;
 
       if (filename) {
-        firebase.storage().ref('avatars')
-          .child(filename).getDownloadURL()
-          .then(this.handleGetUrlSuccess);
+        if (this.oldFilename !== filename) {
```

Comment:
Yes, I agree, it will make the code cleaner! Thank you! 
Note that then the if-else construction should be adapted a little bit as one if and nested ifs is not the same! ;)

I do not agree about the this.oldFilename !== filename as it is only the check of repeated props. It is needed only to do not fetch the filename for every rerender if it is already fetched. Please, try to explain me better if I am wrong here.

Reply:
Now I understand :)
I thought that this condition needed for checking filename when user want to change avatar.